src/Workspaces/Core/Portable/FindSymbols/FindReferences/Finders/AbstractReferenceFinder.cs

```diff
@@ -135,6 +160,22 @@ protected Task<ImmutableArray<Document>> FindDocumentsAsync(Project project, IIm
         protected static bool IdentifiersMatch(ISyntaxFactsService syntaxFacts, string name, SyntaxToken token)
             => syntaxFacts.IsIdentifier(token) && syntaxFacts.TextMatch(token.ValueText, name);
 
+        private static readonly ConditionalWeakTable<ISymbol, string> s_documentationCommentIdCache =
+            new ConditionalWeakTable<ISymbol, string>();
+        private static readonly ConditionalWeakTable<ISymbol, string>.CreateValueCallback s_documentationCommentIdCreateValueCallback =
+            new ConditionalWeakTable<ISymbol, string>.CreateValueCallback(DocumentationCommentId.CreateDeclarationId);
+        protected static bool ShouldFindReferencesInGlobalSuppressions(ISymbol symbol, out Func<string> getDocumentationCommentId)
+        {
+            if (!SupportsGlobalSuppression(symbol))
+            {
+                getDocumentationCommentId = null;
+                return false;
+            }
+
+            getDocumentationCommentId = () => s_documentationCommentIdCache.GetValue(symbol, s_documentationCommentIdCreateValueCallback);
+            return true;
```

Comment:
feels unfortuante we need to do this. also, i believe we have some sort of FindReferenceCache object.  Consider using that so we can cache for the lifetime of FAR< but then release at the end.

Reply:
I am actually not sure if this cache is needed. It seems it would not be expensive to compute documentation comment ID for each top level symbol on which FAR is invoked. I added this cache just to get an idea from others if we feel this is not true. Let me remove this cache for now, and we can add it back if perf shows otherwise.

Reply:
yeah, i think we should remove this and just compute this value on demand.